src/main/scala/dxWDL/compiler/Native.scala

```diff
@@ -106,40 +106,85 @@ case class Native(dxWDLrtId: Option[String],
-                  Some(IR.PARAM_META_PATTERNS -> JsArray(patterns.map(JsString(_))))
-                // If we have the alternative patterns object, extrac the values, if any at all
-                case IR.PatternsReprObj(name, klass, tags) =>
-                  val attrs: Map[String, JsValue] = List(
-                      if (name.isDefined) Some("name" -> JsArray(name.get.map(JsString(_))))
-                      else None,
-                      if (tags.isDefined) Some("tag" -> JsArray(tags.get.map(JsString(_))))
-                      else None,
-                      if (klass.isDefined) Some("class" -> JsString(klass.get)) else None
-                  ).flatten.toMap
-                  // If all three keys for the object version of patterns are None, return None
-                  if (attrs.isEmpty) None else Some(IR.PARAM_META_PATTERNS -> JsObject(attrs))
+        help: Option[Vector[IR.IOAttr]]): Map[String, JsValue] = help match {
+      case None => Map.empty
+      case Some(attributes) => {
+        attributes.flatMap {
+          case IR.IOAttrGroup(text) =>
+            Some(IR.PARAM_META_GROUP -> JsString(text))
+          case IR.IOAttrHelp(text) =>
+            Some(IR.PARAM_META_HELP -> JsString(text))
+          case IR.IOAttrLabel(text) =>
+            Some(IR.PARAM_META_LABEL -> JsString(text))
+          case IR.IOAttrPatterns(patternRepr) =>
+            patternRepr match {
+              case IR.PatternsReprArray(patterns) =>
+                Some(IR.PARAM_META_PATTERNS -> JsArray(patterns.map(JsString(_))))
+              // If we have the alternative patterns object, extrac the values, if any at all
+              case IR.PatternsReprObj(name, klass, tags) =>
+                val attrs: Map[String, JsValue] = List(
+                    if (name.isDefined) Some("name" -> JsArray(name.get.map(JsString(_))))
+                    else None,
+                    if (tags.isDefined) Some("tag" -> JsArray(tags.get.map(JsString(_))))
+                    else None,
+                    if (klass.isDefined) Some("class" -> JsString(klass.get)) else None
+                ).flatten.toMap
+                // If all three keys for the object version of patterns are None, return None
+                if (attrs.isEmpty) None else Some(IR.PARAM_META_PATTERNS -> JsObject(attrs))
+            }
+          case IR.IOAttrChoices(choices) =>
+            Some(IR.PARAM_META_CHOICES -> JsArray(choices.map(choice => {
+              choice match {
+                case IR.ChoiceReprString(value) => JsString(value)
+                case IR.ChoiceReprInteger(value) => JsNumber(value)
+                case IR.ChoiceReprFloat(value) => JsNumber(value)
+                case IR.ChoiceReprBoolean(value) => JsBoolean(value)
+                case IR.ChoiceReprFile(value, name) => {
+                  // TODO: support project and record choices
+                  val dxLink = DxPath.resolveDxURLFile(value).getLinkAsJson
```

Comment:
Nice. Resolving the files is important, because it runtime the applet/app may not have the necessary permissions.